File: renamer.py

import os
import math
import time
import uuid

class Renamer:
    def __init__(self):
        pass

    def calculate_padding(self, total_files):
        """
        根据文件总数计算填充位数。
        <= 99: 2位
        > 99: 3位 (根据需求描述，>99 采用三位数，即 100, 101... 虽然需求没说 >999 怎么办，但通常逻辑是取 max(2, len(str(count))) )
        但严格按照需求：
        "当选择文件数量≤99个时，采用两位数编号格式：01、02、03...99"
        "当选择文件数量>99个时，采用三位数编号格式：001、002、003..."
        这里假设如果超过 999，会自动扩展，或者用户默认不处理那么多。
        为稳健起见，我们使用 max(2, 3 if total_files > 99 else 2) 
        或者更智能一点: len(str(total_files))?
        需求明确指定了规则，我们按规则办。
        <=99 -> 02d
        >99 -> 03d
        """
        if total_files <= 99:
            return 2
        else:
            return 3

    def rename_files(self, file_paths):
        """
        执行重命名操作。
        :param file_paths: 文件绝对路径列表
        """
        if not file_paths:
            return

        total = len(file_paths)
        padding = self.calculate_padding(total)
        
        # 预检查：确保文件存在
        valid_files = [f for f in file_paths if os.path.exists(f)]
        
        try:
            # 直接重命名，一步到位
            for index, old_path in enumerate(valid_files, 1):
                dirname = os.path.dirname(old_path)
                basename = os.path.basename(old_path)
                # 获取扩展名 (保留原始后缀)
                _, ext = os.path.splitext(basename)
                
                # 生成新文件名
                new_name = f"{index:0{padding}d}{ext}"
                new_path = os.path.join(dirname, new_name)
                
                # 避免覆盖自身（如果名字没变）
                if new_path != old_path:
                    os.rename(old_path, new_path)
                
        except Exception as e:
            # 静默失败，不显示任何提示
            pass

File: rust-rename/src/renamer.rs
//! 重命名模块 - 执行批量重命名操作
//!
//! 重命名规则：
//! - 文件数量 <= 99: 使用两位数编号 (01, 02, ...)
//! - 文件数量 > 99: 使用三位数编号 (001, 002, ...)
//! - 保留原始文件扩展名

use std::fs;
use std::path::Path;
use uuid::Uuid;

/// 重命名器
pub struct Renamer;

impl Renamer {
    /// 计算编号的填充位数
    fn calculate_padding(total_files: usize) -> usize {
        if total_files <= 99 { 2 } else { 3 }
    }

    /// 执行批量重命名
    /// 使用两步重命名法防止文件名冲突
    pub fn rename_files(file_paths: &[String]) -> Result<(), String> {
        if file_paths.is_empty() {
            return Ok(());
        }

        // 过滤出存在的文件
        let valid_files: Vec<&String> = file_paths
            .iter()
            .filter(|f| Path::new(f).exists())
            .collect();

        if valid_files.is_empty() {
            return Ok(());
        }

        let total = valid_files.len();
        let padding = Self::calculate_padding(total);

        // 第一步：重命名为临时 UUID 名称
        let mut temp_map: Vec<(String, String)> = Vec::with_capacity(total);

        for old_path in &valid_files {
            let path = Path::new(old_path);
            let dir = path.parent().unwrap_or(Path::new("."));
            let ext = path
                .extension()
                .map(|e| format!(".{}", e.to_string_lossy()))
                .unwrap_or_default();

            let temp_name = format!("{}{}", Uuid::new_v4().to_string().replace("-", ""), ext);
            let temp_path = dir.join(&temp_name);

            fs::rename(old_path, &temp_path).map_err(|e| {
                format!("重命名 {} 到临时文件失败: {}", old_path, e)
            })?;

            temp_map.push((temp_path.to_string_lossy().to_string(), ext));
        }

        // 第二步：重命名为最终编号名称
        for (index, (temp_path, ext)) in temp_map.iter().enumerate() {
            let path = Path::new(temp_path);
            let dir = path.parent().unwrap_or(Path::new("."));

            let new_name = format!("{:0width$}{}", index + 1, ext, width = padding);
            let new_path = dir.join(&new_name);

            fs::rename(temp_path, &new_path).map_err(|e| {
                format!("重命名临时文件到 {} 失败: {}", new_path.display(), e)
            })?;
        }

        Ok(())
    }
}
